src/vs/workbench/browser/parts/editor/editorTabsControl.ts

```diff
@@ -317,15 +339,15 @@ export abstract class EditorTabsControl extends Themable {
 		const currentResourceContext = this.resourceContext.get();
 		this.resourceContext.set(EditorResourceAccessor.getOriginalUri(editor, { supportSideBySide: SideBySideEditor.PRIMARY } ?? null));
 		const currentPinnedContext = !!this.editorPinnedContext.get();
-		this.editorPinnedContext.set(this.group.isPinned(editor));
+		this.editorPinnedContext.set(this.tabsModel.isPinned(editor));
```

Comment:
I see some inconsistent use of `tabsModel` vs. `groupViewer` in this file. I would expected all class that can be made on `tabsModel` to be made on it.

Reply:
Not all lookups can be done on the tabsModel as we require the global view in many cases. For example for the listeners and to get the "real" editor index